File: kotlin/toolbar_TabLayout_Viewpage2_navigation/app/src/main/java/com/funshow/toolbar_tablayout_viewpage2_navigation/FragmentViewPagerAdapter.java
package com.funshow.toolbar_tablayout_viewpage2_navigation;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class FragmentViewPagerAdapter extends FragmentStateAdapter {
    //初始化
    public FragmentViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 0){
            return new FragmentA();
        }
        else if (position == 1){
            return new FragmentB();
        }
        else {
            return new FragmentC();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }
}

File: kotlin/toolbar_tablayout_viewpage2_navigation2/app/src/main/java/com/funshow/toolbar_tablayout_viewpage2_navigation/FragmentViewPagerAdapter.kt
package com.funshow.toolbar_tablayout_viewpage2_navigation

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class FragmentViewPagerAdapter  //初始化
    (fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return if (position == 0) {
            FragmentA()
        } else if (position == 1) {
            FragmentB()
        } else {
            FragmentC()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}
